internal/lsp/completions/providers/rulerefs.go

```diff
@@ -33,108 +36,159 @@ func (*RuleFromImportedPackageRefs) Run(
 		return nil, nil
 	}
 
+	words := patternWhiteSpace.Split(currentLine, -1)
+	lastWord := words[len(words)-1]
+
 	// some version of a parsed mod is needed here to filter refs to suggest
 	// based on import statements
 	mod, ok := c.GetModule(fileURI)
 	if !ok {
 		return nil, nil
 	}
 
-	refsFromImports := make(map[string]types.Ref)
+	refsForContext := make(map[string]types.Ref)
 
-	for file, refs := range c.GetAllFileRefs() {
-		if file == fileURI {
-			continue
-		}
-
-		for key, ref := range refs {
-			// we are not interested in packages here, only the rules
+	for uri, refs := range c.GetAllFileRefs() {
+		for _, ref := range refs {
+			// we are not interested in packages here, only rules
```

Comment:
I guess there are a few cases where you might want to refer to a package even in a rule body, like when doing dynamic policy composition:

```
some policy
some decision in data.policies[policy].deny
```
and so on... but I guess that's uncommon enough that it's still better to only suggest rules as a default.

Reply:
Hmm, that is a thought. I guess this provide is called rulers and is already quite complex. I'd like to consider that for a new provider.

Issue here: https://github.com/StyraInc/regal/issues/771